### PR TITLE
[2159] schools snagging nonjs form

### PR DESCRIPTION
**Pull request by stellenberger**
### Context

- [This trello ticket](https://trello.com/c/aQw0KyVw/2159-schools-snagging-nonjs-form)

### Changes proposed in this pull request

- Changed the error message if a user does not select any radio button on the Lead and Employing school NonJS form
- Fixed the `_results` partial to use the `legend` and `hint` functionalities in the `govuk_radio_buttons_fieldset` in both Lead and Employing school forms

### Guidance to review

- Make a trainee on school direct salaried
- Turn off JS and input at least 3 valid chars.
- On the radio button and NonJS page, dont select anything. You will see a custom error message saying `Select an lead/employing school or search again` (lead/employing dependent on which form you are on)

### CODE REVIEW

**Comment by stellenberger** [comment]
[Schools snagging - nonJS form](https://trello.com/c/aQw0KyVw/2159-schools-snagging-nonjs-form)

File: spec/support/shared_examples/school_form_validations.rb [hunk 1]
```diff
@@ -18,12 +18,4 @@
       expect(subject.errors[:results_search_again_query]).to include(I18n.t("activemodel.errors.models.schools_form.attributes.query.length", count: 2))
     end
   end
-
```
Comment (jhenderson) [inline spec/support/shared_examples/school_form_validations.rb:21]: I don't think you need to delete this. Happened do this in my PR, but you might as well do it here now:

```ruby
context "with no school chosen and no query provided" do
  let(:form_name) { school_id_key.sub("id", "form") }
  let(:params) { { "results_search_again_query" => "", school_id_key => "" } }

  it "returns an error" do
    expect(subject.errors[school_id_key]).to include(
      I18n.t("activemodel.errors.models.schools/#{form_name}.attributes.#{school_id_key}.blank"),
    )
  end
end
```

**Review by jhenderson — CHANGES_REQUESTED** [review]

File: spec/support/shared_examples/school_form_validations.rb (hunk 1 above)
Comment (stellenberger) [reply spec/support/shared_examples/school_form_validations.rb:21]: All done 👍

**Review by jhenderson — APPROVED** [review]

**Review by edwardhorsford — APPROVED** [review]

**Comment by bbelward** [comment]
Tested and is working as expected. Have closed the [card](https://trello.com/c/aQw0KyVw) on Trello 👍

**Pull request merged**